crates/recent_projects/src/remote_servers.rs

```diff
@@ -589,23 +594,32 @@ impl RemoteServerProjects {
                         });
                     };
 
-                    let project = cx.update(|_, cx| {
-                        project::Project::ssh(
-                            session,
-                            app_state.client.clone(),
-                            app_state.node_runtime.clone(),
-                            app_state.user_store.clone(),
-                            app_state.languages.clone(),
-                            app_state.fs.clone(),
-                            cx,
+                    let (path_style, project) = cx.update(|_, cx| {
+                        (
+                            session.read(cx).path_style(),
+                            project::Project::ssh(
+                                session,
+                                app_state.client.clone(),
+                                app_state.node_runtime.clone(),
+                                app_state.user_store.clone(),
+                                app_state.languages.clone(),
+                                app_state.fs.clone(),
+                                cx,
+                            ),
                         )
                     })?;
 
                     let home_dir = project
                         .read_with(cx, |project, cx| project.resolve_abs_path("~", cx))?
                         .await
                         .and_then(|path| path.into_abs_path())
-                        .unwrap_or(PathBuf::from("/"));
+                        .map(|path| RemotePathBuf::new(path, path_style))
+                        .unwrap_or_else(|| match path_style {
+                            PathStyle::Posix => RemotePathBuf::from_str("/", PathStyle::Posix),
```

Comment:
NIT: This also looks like a method for `PathStyle`, esp. if we use it more than once around.